data_model/src/lib.rs

```diff
@@ -355,6 +357,19 @@ impl Value {
     }
 }
 
+impl<A: small::Array> From<SmallVec<A>> for Value
+where
+    A::Item: Into<Value>,
+{
```

Comment:
do we still need `From<Vec<V>> for Value`?
and what about `TryFrom<Value> for SmallVec<A>`?

Reply:
> do we still need From<Vec<V>> for Value?

Yes. We can't replace `Vec` everywhere, because `SmallVec<Value>` would be recursive, if `Value::Vec` wasn't heap-allocated, and `Box` defeats the purpose. 

> TryFrom<Value> for SmallVec<A>

Good point. Will add shortly.